### PR TITLE
Added docs for blitz connection_alias and harness support for differe…

### PR DESCRIPTION
**Pull request by lsheikal**
Added Docs for the following:

1. Enhancement for Blitz actions execute, parse and configure.
2. Enhancement for genie harness

### CODE REVIEW

File: docs/blitz/design/actions/actions_list.rst [hunk 1]
```diff
@@ -16,6 +16,8 @@ exists in the output. You also, have the option to check if a specific
         device: R1
         # Send show version to the device
         command: show version
+        # To execute command in different connections, by default cli
```
Comment (tahigash) [inline docs/blitz/design/actions/actions_list.rst:19]:
```suggestion
        # To execute command with specified connection alias, 'cli' is default
```

File: docs/blitz/design/actions/actions_list.rst [hunk 2]
```diff
@@ -26,7 +28,12 @@ exists in the output. You also, have the option to check if a specific
             - 'Should not be in the output'
 
 
-Both include and exclude keywords are optional to use.
+Both include, exclude and connection_alias keywords are optional to use.
```
Comment (tahigash) [inline docs/blitz/design/actions/actions_list.rst:31]:
```suggestion
include, exclude and connection_alias keywords are optional to use.
```

File: docs/blitz/design/actions/actions_list.rst [hunk 3]
```diff
@@ -26,7 +28,12 @@ exists in the output. You also, have the option to check if a specific
...
+But to use connection_alias two things needs to be followed:
```
Comment (tahigash) [inline docs/blitz/design/actions/actions_list.rst:32]:
```suggestion
Two things needs to be followed to use connection_alias:
```

File: docs/blitz/design/actions/actions_list.rst [hunk 4]
```diff
@@ -26,7 +28,12 @@ exists in the output. You also, have the option to check if a specific
...
+    1. New mapping datafile schema.
+    2. ``alias`` in mapping datafile should be the ``connection_alias``.
+How to use the new schema for different connection can be found at this `link
+<https://pubhub.devnetcloud.com/media/genie-docs/docs/cookbooks/harness.html#what-can-you-do-with-the-mapping-datafile>`__.
```
Comment (tahigash) [inline docs/blitz/design/actions/actions_list.rst:36]: just comment. very good to have pointer like this!

File: docs/blitz/design/actions/actions_list.rst [hunk 5]
```diff
@@ -51,10 +58,14 @@ The `configure` action is used to configure the device.
 
     - configure: # ACTION
         device: device_name
+        # To configure using different connections, by default cli
```
Comment (tahigash) [inline docs/blitz/design/actions/actions_list.rst:61]:
```suggestion
        # To execute command with specified connection alias, 'cli' is default
```

File: docs/cookbooks/harness.rst [hunk 6]
```diff
@@ -631,6 +649,7 @@ information such as:
 2. Provide connection pool information per device
 3. Overwrite devices alias
 4. Which connection type to prioritize for the run
+5. Support for custom connections
```
Comment (tahigash) [inline docs/cookbooks/harness.rst:652]:
```suggestion
5. Support for multiple connections per context(cli/yang/etc) with new schema
```

File: docs/userguide/harness/user/datafile.rst [hunk 7]
```diff
@@ -709,6 +709,66 @@ they have either;
 
                     label: # Label to override th alias of the specified interface
 
+
+* Different connections using Mapping datafile:
+
+The user can connect to custom connections using this new schema mentioned below.
```
Comment (tahigash) [inline docs/userguide/harness/user/datafile.rst:715]:
```suggestion
The user can connect with different connections using this new schema mentioned below.
```

File: docs/userguide/harness/user/datafile.rst [hunk 8]
```diff
@@ -709,6 +709,66 @@ they have either;
...
+                label:  # One device in the testbed must be designated as the `uut`
+                        # for ``Genie``.  If no device is already named `uut`, then
+                        # you must map it via `label` field, as shown in the example
+                        # above. In the event that the device is already named uut
+                        # via `alias` in the testbed file, then there is no need to
+                        # use label field.
+
+              mapping: # To map which connection to use for in the context.
+
+              context: # Context for the Genie execution.
```
Comment (tahigash) [inline docs/userguide/harness/user/datafile.rst:758]: indentation
```suggestion
              mapping: # To map which connection to use for in the context.

                  context: # Context for the Genie execution.
```

**Review by tahigash — CHANGES_REQUESTED** [review]

File: docs/cookbooks/harness.rst [hunk 9]
```diff
@@ -327,6 +327,24 @@ The mapping datafile must be modified to add the pool size.
 This will initialize 2 connections to the uut and the cli commands will go up
 to twice as fast! This number can scale up to what your device can support.
 
+In order to use connection pool for different connections, the new mapping schema needs to be followed.
```
Comment (dwapstra) [inline docs/cookbooks/harness.rst:330]: Do we support old and new schemas? How does the user know which schema is used? If there is only a single schema, we can drop the 'new mapping schema' nomenclature and just describe what is the required structure.

File: docs/cookbooks/harness.rst (hunk 6 above)
Comment (dwapstra) [reply docs/cookbooks/harness.rst:652]: Do we need the 'with new schema' ?

File: docs/cookbooks/harness.rst [hunk 10]
```diff
@@ -650,6 +669,22 @@ datafile.
              cli: vty
              yang: yang
 
+The custom connections are supported in mapping datafile, by the new schema:
```
Comment (dwapstra) [inline docs/cookbooks/harness.rst:672]: in line with other comments. custom -> multiple.  Not sure if we need 'new schema'

File: docs/blitz/design/actions/actions_list.rst (hunk 1 above)
Comment (lsheikal) [reply docs/blitz/design/actions/actions_list.rst:19]: updated

File: docs/blitz/design/actions/actions_list.rst (hunk 2 above)
Comment (lsheikal) [reply docs/blitz/design/actions/actions_list.rst:31]: updated

File: docs/blitz/design/actions/actions_list.rst (hunk 3 above)
Comment (lsheikal) [reply docs/blitz/design/actions/actions_list.rst:32]: updated

File: docs/blitz/design/actions/actions_list.rst (hunk 4 above)
Comment (lsheikal) [reply docs/blitz/design/actions/actions_list.rst:36]: Thank you takashi

File: docs/blitz/design/actions/actions_list.rst (hunk 5 above)
Comment (lsheikal) [reply docs/blitz/design/actions/actions_list.rst:61]: updated

File: docs/cookbooks/harness.rst (hunk 6 above)
Comment (lsheikal) [reply docs/cookbooks/harness.rst:652]: updated @tahigash. @dwapstra Kindly let me know if there is a change needed.

File: docs/userguide/harness/user/datafile.rst (hunk 7 above)
Comment (lsheikal) [reply docs/userguide/harness/user/datafile.rst:715]: updated

File: docs/userguide/harness/user/datafile.rst (hunk 8 above)
Comment (lsheikal) [reply docs/userguide/harness/user/datafile.rst:758]: updated

File: docs/cookbooks/harness.rst (hunk 9 above)
Comment (lsheikal) [reply docs/cookbooks/harness.rst:330]: yes @dwapstra we support both new and old schema.

File: docs/cookbooks/harness.rst (hunk 10 above)
Comment (lsheikal) [reply docs/cookbooks/harness.rst:672]: updated custom to multiple connections.

**Review by tahigash — APPROVED** [review]
thanks for addressing comments!

File: docs/userguide/harness/user/datafile.rst [hunk 11]
```diff
@@ -709,6 +709,66 @@ they have either;
 
                     label: # Label to override th alias of the specified interface
 
+
+* Different connections using Mapping datafile:
+
+The user can connect with different connections using this new schema mentioned below.
+
+``Note``: The user can either use this new schema or the above mentioned one.
```
Comment (domachad) [inline docs/userguide/harness/user/datafile.rst:717]:
```suggestion
``Note``: The user can either use this new schema or the one mentioned above.
```

File: docs/userguide/harness/user/datafile.rst [hunk 12]
```diff
@@ -709,6 +709,66 @@ they have either;
 
                     label: # Label to override th alias of the specified interface
 
+
+* Different connections using Mapping datafile:
+
+The user can connect with different connections using this new schema mentioned below.
```
Comment (domachad) [inline docs/userguide/harness/user/datafile.rst:715]:
```suggestion
The user can connect with different connections using the following schema:
```

File: docs/userguide/harness/user/datafile.rst [hunk 13]
```diff
@@ -709,6 +709,66 @@ they have either;
...
+                        # via `alias` in the testbed file, then there is no need to
+                        # use label field.
+
+              mapping: # To map which connection to use for in the context.
+
+                  context: # Context for the Genie execution.
+
+                            via: #Specify the connection to use. Eg. cli, ssh, yang etc
+
+                            alias: #The alias name of the connection. Eg. Banana, my_conn
```
Comment (domachad) [inline docs/userguide/harness/user/datafile.rst:762]:
```suggestion
                            alias: #The alias name of the connection. Eg. my_connection, my_conn
```

**Review by domachad — APPROVED** [review]
Approved with suggestions

File: docs/userguide/harness/user/datafile.rst (hunk 11 above)
Comment (lsheikal) [reply docs/userguide/harness/user/datafile.rst:717]: updated

File: docs/userguide/harness/user/datafile.rst (hunk 12 above)
Comment (lsheikal) [reply docs/userguide/harness/user/datafile.rst:715]: updated

File: docs/userguide/harness/user/datafile.rst (hunk 13 above)
Comment (lsheikal) [reply docs/userguide/harness/user/datafile.rst:762]: updated

**Pull request merged**